rotkehlchen/chain/evm/decoding/extrafi/decoder.py

```diff
@@ -0,0 +1,392 @@
+                address=context.tx_log.address,
+                counterparty=CPT_EXTRAFI,
+            )
+
+        for event in context.decoded_events:
+            if (
+                event.maybe_get_direction() == EventDirection.OUT and
+                event.location_label == manager and
+                (
+                    event.balance.amount == asset_normalized_value(amount_0_invested, event.asset) or  # type: ignore  # noqa: E501
+                    event.balance.amount == asset_normalized_value(amount_1_invested, event.asset)  # type: ignore
+                )
+            ):
+                event.counterparty = CPT_EXTRAFI
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.extra_data = {
+                    'vault_id': vault_id,
+                    'vault_position': hex_or_bytes_to_int(context.tx_log.topics[2]),
+                }
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol_or_name()} in Extrafi {self._farm_name(vault_id)}'  # noqa: E501
+                if amount_0_invested == 0 or amount_1_invested == 0:
+                    # It means we have decoded the only existing event and we can exit the logic
+                    break
+        else:
+            log.error(f'Could not find the invest event for extrafi in {context.transaction}')
+
+        return DecodingOutput(borrow_event)
+
+    def _handle_farm_repayment(self, context: DecoderContext) -> DecodingOutput:
+        amount_0_repaid = hex_or_bytes_to_int(context.tx_log.data[32:64])
+        amount_1_repaid = hex_or_bytes_to_int(context.tx_log.data[64:96])
+        vault_id = hex_or_bytes_to_int(context.tx_log.topics[1])
+        refund_event: EvmEvent | None = None
+
+        # the refund event appears before the send event in the case of an ETH transfer
+        # or after the send event in the case of tokens
+        for event in context.decoded_events[1:]:
+            if event.maybe_get_direction() == EventDirection.IN:
+                refund_event = event
+                break
+
+        for event in context.decoded_events[1:]:
+            if event.maybe_get_direction() != EventDirection.OUT or event.counterparty == CPT_GAS:
+                continue
+
+            comparing_amount = event.balance.amount
+            if (
+                refund_event is not None and
+                event.asset == refund_event.asset and
+                event.address == refund_event.address
+            ):
+                # When there is a refund in the transaction we need to check that the amount
+                # matches taking into account that we sent a little bit more
+                comparing_amount -= refund_event.balance.amount
+
+            if not (
+                comparing_amount == asset_normalized_value(amount_0_repaid, event.asset) or  # type: ignore
```

Comment:
type ignore without a comment should not be done. Ever. Why did you ignore the type here? Seems to be because event.asset is Asset and here it's CryptoAsset you expect.

Shouldn't you resolve it? Same for next line.

Reply:
I did it because the function resolves it... we can improve this in other function since I believe we are double resolving many times